src/Gui/NaviCube.cpp

```diff
@@ -416,24 +415,36 @@ GLuint NaviCubeImplementation::createCubeFaceTex(QtGLWidget* gl, float gap, cons
 	}
 	else if (shape == SHAPE_SQUARE) {
 		QPainterPath pathSquare;
-		pathSquare.addRect(QRectF(gapi, gapi, (qreal)texSize - 2.0 * gapi, (qreal)texSize - 2.0 * gapi));
+		auto rectSquare = QRectF(gapi, gapi, (qreal)texSize - 2.0 * gapi, (qreal)texSize - 2.0 * gapi);
+		// Qt's coordinate system is x->left y->down, this must be taken into account on operations
+		pathSquare.moveTo(rectSquare.left()         , rectSquare.bottom() - gapi);
+		pathSquare.lineTo(rectSquare.left() + gapi  , rectSquare.bottom());
+		pathSquare.lineTo(rectSquare.right() - gapi , rectSquare.bottom());
+		pathSquare.lineTo(rectSquare.right()        , rectSquare.bottom() - gapi);
+		pathSquare.lineTo(rectSquare.right()        , rectSquare.top() + gapi);
+		pathSquare.lineTo(rectSquare.right() - gapi , rectSquare.top());
+		pathSquare.lineTo(rectSquare.left() + gapi  , rectSquare.top());
+		pathSquare.lineTo(rectSquare.left()         , rectSquare.top() + gapi);
+		pathSquare.lineTo(rectSquare.left()         , rectSquare.bottom() - gapi);
 		paint.fillPath(pathSquare, Qt::white);
 	}
 	else if (shape == SHAPE_CORNER) {
 		QPainterPath pathCorner;
-		QRectF rectCorner = QRectF(3.46 * gapi, 3.31 * gapi, sqrt(2) * gapi, 1.3 * gapi);
-		pathCorner.moveTo(rectCorner.left() + (rectCorner.width() / 2), rectCorner.top());
-		pathCorner.lineTo(rectCorner.bottomLeft());
-		pathCorner.lineTo(rectCorner.bottomRight());
-		pathCorner.lineTo(rectCorner.left() + (rectCorner.width() / 2), rectCorner.top());
+		auto rectCorner = QRectF((texSize - 2 * sqrt(2) * gapi) / 2, (texSize - sqrt(6) * gapi) / 2, 2 * sqrt(2) * gapi, sqrt(6) * gapi);
```

Comment:
Very nice work on this PR, thanks to both @adrianinsaval and @Rexbas for your contributions. I appreciate the move away from "magic numbers" here -- can you add a (short) comment explaining here why we are using sqrt(6), which otherwise still looks a bit magic? Also below, the 0.25 and 0.75 probably warrant explanation. Thanks!

Reply:
sqrt(6) is the height of a hexagon with a side length of sqrt(2)

Reply:
Suggestion to skip the  `rectCorner` and calculate the horisontal and vertical positins of the hexagons corner. May make more sense:
```
// The horisontal corner positions of the hexagon
float hexWidth = 2 * sqrt(2) * gapi
float hexX0 = (texSize - hexWidth) / 2
float hexX1 = hexX0 + hexWidth * 0.25
float hexX2 = hexX0 + hexWidth * 0.75
float hexX3 = hexX0 + hexWidth

// The vertical corner positions of the hexagon
float hexHeight = sqrt(6) * gapi
float hexY0 = (texSize - hexHeight) / 2
float hexY1 = hexX0 + hexHeight * 0.5
float hexY2 = hexX0 + hexHeight

// Draw hexagon
QPainterPath pathCorner;
pathCorner.moveTo(hexX0, hexY1);
pathCorner.lineTo(hexX1, hexY2);
pathCorner.lineTo(hexX2, hexY2);
pathCorner.lineTo(hexX3, hexY1);
pathCorner.lineTo(hexX2, hexY0);
pathCorner.lineTo(hexX1, hexY0);
pathCorner.closeSubpath();

```

Reply:
@chennes do you prefer what I pushed now or should I change to what Jolbas proposed?

Reply:
I leave that to you: I am happy to merge as-is, and @Jolbas can then submit a PR with the refactor, or you can incorporate the change now. Just let me know what you want to do.

Reply:
I am happy with it as is, just pushed a typo fix reported by CI now